packages/@lwc/synthetic-shadow/src/polyfills/window-event-target/polyfill.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { windowRemoveEventListener, windowAddEventListener } from '../../env/window';
+
+import { defineProperties, isUndefined, isTrue, ArraySlice } from '@lwc/shared';
+import { eventTargetGetter } from '../../env/dom';
+import { isNodeShadowed } from '../../shared/node-ownership';
+
+// this method returns true if an event can be seen by a particular eventTarget,
+// otherwise it returns false, which means the listener will never be invoked.
+function isQualifyingEventTarget(currentTarget: Window, evt: Event): boolean {
+    const { composed } = evt;
+    const originalTarget = eventTargetGetter.call(evt);
+    if (originalTarget === currentTarget) {
+        // dispatched on the window directly
+        return true;
+    }
+    // if the event reaches the window by propagation is because it is bubbling,
```

Comment:
```suggestion
    // if the event reaches the window by propagation, it is because it is bubbling.
    // In which case, it really depends on the event's composed value
```